crates/zk-por-core/src/circuit_registry/registry.rs

```diff
@@ -0,0 +1,167 @@
+use plonky2::plonk::{
+    circuit_data::{CircuitConfig, CircuitData},
+    proof::ProofWithPublicInputs,
+};
+
+use crate::{
+    account::Account,
+    // circuit_config::{BATCH_SIZE, RECURSIVE_FACTOR, ASSET_NUM},
+    merkle_sum_prover::{
+        circuits::{
+            account_circuit::AccountTargets, merkle_sum_circuit::build_merkle_sum_tree_circuit,
+        },
+        prover::MerkleSumTreeProver,
+    },
+    recursive_prover::prover::RecursiveProver,
+    recursive_prover::recursive_circuit::{build_recursive_n_circuit, RecursiveTargets},
+};
+
+use plonky2_field::types::Field;
+use rand::Rng;
+
+use crate::types::{C, D, F};
+// use once_cell::sync::Lazy;
+
+pub struct CircuitRegistry<const RECURSIVE_FACTOR: usize> {
+    batch_circuit: (CircuitData<F, C, D>, Vec<AccountTargets>),
+    empty_batch_proof: ProofWithPublicInputs<F, C, D>,
+    recursive_circuits_and_empty_proofs: Vec<(
```

Comment:
if we use digest to index; we can use a map; then, can separate circuits and empty_proofs

Reply:
no need to use digest to index.